FastTimeSeries/src/models/StackedLSTM.jl

```diff
@@ -0,0 +1,50 @@
+# StackedLSTM.jl
+#
+# Layers for stacked LSTM (referenced from https://github.com/sdobber/FluxArchitectures.jl)
+
+mutable struct StackedLSTMCell{A}
+	chain::A
+end
+
+"""
+	StackedLSTM(in, out, hiddensize, layers)
+
+Stacked LSTM network. Feeds the data through a chain of LSTM layers, where the hidden state
+of the previous layer gets fed to the next one. The first layer corresponds to
+`LSTM(in, hiddensize)`, the hidden layers to `LSTM(hiddensize, hiddensize)`, and the final
+layer to `LSTM(hiddensize, out)`. Takes the keyword argument `init` for the initialization
+of the layers.
+
+"""
+function StackedLSTM(in::Int, out::Integer, hiddensize::Integer, layers::Integer;
+			init=Flux.glorot_uniform)
+	if layers == 1
+		chain = Chain(LSTM(in, out; init=init))
+	elseif layers == 2
+		chain = Chain(LSTM(in, hiddensize; init=init),
+					  LSTM(hiddensize, out; init=init))
+	else
+		chain_vec = [LSTM(in, hiddensize; init=init)]
+		for i = 1:layers - 2
+			push!(chain_vec, LSTM(hiddensize, hiddensize; init=init))
+		end
+		chain = Chain(chain_vec..., LSTM(hiddensize, out; init=init))
+	end
+	return StackedLSTMCell(chain)
+end
+
+function (m::StackedLSTMCell)(X)
+	[m.chain(x) for x ∈ X[1:end-1]]
+	return m.chain(X[end])
```

Comment:
Is this always a sequence-to-one model?

Reply:
I think for classification and regression, this will always be a seq-to-one model ?
For some nlp tasks, it would not be.

Reply:
Ok, then I think you should be able to rewrite this using `foldl`. It'll be more correct and likely perform better as well. It may also let you drop the `Recur` wrapper for the inner layer stack and use the cells directly instead (i.e. `LSTM -> LSTMCell`).